larq_compute_engine/core/bgemm_impl_ruy.h

```diff
@@ -53,8 +53,9 @@ struct BGemmImplUsingRuy {
     dst.data = dst_data;
 
     TSpec spec;
-    spec.fused_multiply = params.fused_multiply;
-    spec.fused_add = params.fused_add;
+    spec.backtransform_add = params.backtransform_add;
```

Comment:
IMO, it would be better to define `pre_actication_BN` stuff and and just fuse the backtransformation into that. In this way, you have implemented pre-actication BN and backtransformation at the same time and there is no need to define explcit  backtranformation attributes  all over the places.

Reply:
I like the `pre_XXX` and `post_XXX` idea, I've been thinking about it too. The `float` vs `int32` is tricky though.
The reason I would rather do the backtransform in `int32` is that it will still work for the `int8` assembly kernels. For `int8`, all the `int32` stuff can stay the same but all the `float` stuff will be *approximated* by these weird exponent+shifts.  So if we change the backtransform to a float, then it would become an approximation in the int8 kernel, and I'd rather keep it as an 'exact' calculation whenever possible.

Reply:
so based on our phone discussion, since we are not planning to support BT+BN(float)+RELU+BN(FLOAT) the float impl. is not relevant then.